### PR TITLE
[CIRC-295] Fix sending recall notice on loan due date changed

### PR DESCRIPTION
**Pull request by dmytrotur**
## Purpose
Fixes behavior for [CIRC-295](https://issues.folio.org/browse/CIRC-295)

## Approach
Changing boolean flag to "old due date" value. 
The reason for this change is that there are cases when due date is changed several times, and the last value is the same as value before all changes.

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.

### CODE REVIEW

File: src/main/java/org/folio/circulation/domain/Loan.java [hunk 1]
```diff
@@ -33,29 +33,20 @@
   private final Item item;
   private final User user;
   private final User proxy;
+  private final DateTime oldDueDate;
```
Comment (marcjohnson-kint) [inline src/main/java/org/folio/circulation/domain/Loan.java]: Maybe we could call this `originalDueDate` to reflect that it was the due date that the loan had when the current process started?

**Comment by marcjohnson-kint** [comment]
How has this case discovered?

This change does not appear to include any new tests to demonstrate the possibility?

File: src/main/java/org/folio/circulation/domain/Loan.java (hunk 1 above)
Comment (dmytrotur) [reply src/main/java/org/folio/circulation/domain/Loan.java]: Makes sense, I'll do this change.

Comment (dmytrotur) [reply src/main/java/org/folio/circulation/domain/Loan.java]: Done

**Comment by dmytrotur** [comment]
One of the existing tests was changed to reflect this case.
The example of such behavior is long-term loans: first due date is calculated as loan date + period, then it is moved to the end of a day (in closed library due date management).

**Comment by marcjohnson-kint** [comment]
> One of the existing tests was changed to reflect this case.

Apologies, I didn't spot the importance of the addition of closed library due date management to that test

> The example of such behavior is long-term loans: first due date is calculated as loan date + period, then it is moved to the end of a day (in closed library due date management).

**Pull request merged**